### PR TITLE
Add Loading Indicator for Data Fetching

### PR DESCRIPTION
**Pull request by lgaleana-llm**
Added a 'Loading...' indicator to the UI while the text and images are being scraped. Updated the HTML template and JavaScript to show the loading text during the data fetching process and hide it once the data is loaded. Added a Playwright test to verify the loading indicator functionality.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
### Summary of Test Failures for `test_ui_loading_indicator`

#### 1. What were the errors?

- **Initial Error**:
  - The test failed because the `page.wait_for_selector("#loading-indicator", state="visible")` timed out after 30 seconds.
  - The log `Loading indicator visible before wait: False` confirmed that the "Loading..." indicator was not visible before the wait.

- **Subsequent Errors**:
  - The same timeout error persisted, indicating that the "Loading..." indicator was not becoming visible within the expected time frame.
  - The fetch request to `/fetch-image` returned a `400 Bad Request` error, which might have been causing the loading indicator to not appear.

#### 2. What did you try to fix them?

- **First Attempt**:
  - Added a short wait after the form submission to give the "Loading..." indicator time to appear.
  - Added logging to verify the state of the "Loading..." indicator before and after the wait.

- **Second Attempt**:
  - Mocked the `/fetch-image` request to return a successful response to avoid the `400 Bad Request` error.
  - Added more detailed logging in the JavaScript code to trace the execution flow and ensure the "Loading..." indicator was being shown.

- **Third Attempt**:
  - Simplified the test to focus solely on the visibility of the "Loading..." indicator.
  - Used `page.evaluate` to ensure the JavaScript code was executed correctly and the "Loading..." indicator was shown.
  - Added more detailed logging in both the JavaScript code and the test to trace the execution flow.

Despite these efforts, the test continued to fail due to the "Loading..." indicator not becoming visible within the expected time frame.

**Comment by lgaleana-llm** [comment]
Total time to create PR: 396.09 seconds
Total OPenAI cost: 2.851065

File: tests/test_ui/test_ui_loading_indicator.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import pytest
```
Comment (lgaleana) [inline tests/test_ui/test_ui_loading_indicator.py]: This change is too simple. No need to test it. Remove this file.

Comment (lgaleana-llm) [reply tests/test_ui/test_ui_loading_indicator.py]: Thank you for the feedback. I understand your concern. I'll remove the test file for the loading indicator as the change is straightforward and doesn't require additional testing.

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 432.00 seconds
Total OPenAI cost: 0.37778

**Review by lgaleana — APPROVED** [review]

**Pull request merged**